src/app_service/service/community/service.nim

```diff
@@ -208,6 +212,18 @@ QtObject:
         var chatDetails = self.chatService.getChatById(fullChatId)
         result.add(chatDetails)
 
+  proc handleCommunitiesSettingsUpdates(self: Service, communitiesSettings: seq[CommunitySettingsDto]) =
+    for settings in communitiesSettings:
+      var emitUpdate = false
+      if self.allCommunities.hasKey(settings.id):
+        self.allCommunities[settings.id].settings = settings
+      if self.joinedCommunities.hasKey(settings.id):
+        self.joinedCommunities[settings.id].settings = settings
+        emitUpdate = true
```

Comment:
@PascalPrecht since you're sending `SIGNAL_COMMUNITY_EDITED` signal in a loop only when `emitUpdate` is true and since this is the only place when you're setting `emitUpdate` to true, maybe you can remove `emitUpdate` completely and just send signal from this "then" block.

Reply:
I think the intention was different. `emitUpdate` should be defined before the loop, signal `SIGNAL_COMMUNITY_EDITED` should be emitted only when the loop is finished.

Reply:
@saledjenic suggestion is correct here. Should move the emission inside the second if block.

Originally I thought about emitting updates in case it only matches for `allCommunities` as well, but actually this is not needed, as we're only interested in community settings from joined communities.

Will update this accordingly.

Thanks for the review guys!

Reply:
Ah yes, sorry `emitUpdate` flag confused me too much, was sure the intention was different. But obviously, SIGNAL_COMMUNITY_EDITED must be emitted per community.